From bd1cd6ae750f91ed651a39a002206d0c9bb80354 Mon Sep 17 00:00:00 2001
From: "Hebbar, Gururaja" <gururaja.hebbar@ti.com>
Date: Tue, 6 Mar 2012 23:41:17 +0530
Subject: [PATCH 0809/1092] ARM: OMAP2+: I2C: hwmod: set flag to restore
 context

Restore of i2c context is not done for am335x due to missing flag. This
patch adds the OMAP_I2C_FLAG_RESET_REGS_POSTIDLE in the am335x
hwmod data which activates the restore.
This also solves the problem shown below of i2c instances not resuming
correctly from suspend. (especially those not in wakeup domain)

[   42.610146] omap_i2c omap_i2c.2: XRDY IRQ while no data to send
[   43.611192] omap_i2c omap_i2c.2: controller timed out

Signed-off-by: Hebbar, Gururaja <gururaja.hebbar@ti.com>
---
 arch/arm/mach-omap2/omap_hwmod_33xx_data.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
index 1c8f252..4f00471 100644
--- a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
+++ b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
@@ -982,7 +982,8 @@ static struct omap_hwmod_class_sysconfig am33xx_i2c_sysc = {
 };
 
 static struct omap_i2c_dev_attr i2c_dev_attr = {
-	.flags		= OMAP_I2C_FLAG_BUS_SHIFT_NONE,
+	.flags		= OMAP_I2C_FLAG_BUS_SHIFT_NONE |
+					OMAP_I2C_FLAG_RESET_REGS_POSTIDLE,
 };
 
 static struct omap_hwmod_class i2c_class = {
-- 
1.7.11.2

